common/config/loader.go

```diff
@@ -90,14 +93,28 @@ func Load(env string, configDir string, zone string, config interface{}) error {
 	// TODO: remove log dependency.
 	stdlog.Printf("Loading config files=%v\n", files)
 
+	templateFuncs := sprig.FuncMap()
+
 	for _, f := range files {
 		// This is tagged nosec because the file names being read are for config files that are not user supplied
 		// #nosec
 		data, err := os.ReadFile(f)
 		if err != nil {
 			return err
 		}
-		err = yaml.Unmarshal(data, config)
+
+		tpl, err := template.New("config").Funcs(templateFuncs).Parse(string(data))
+		if err != nil {
+			return err
+		}
+
+		var rendered bytes.Buffer
+		err = tpl.Execute(&rendered, nil)
+		if err != nil {
+			return err
+		}
+
+		err = yaml.Unmarshal(rendered.Bytes(), config)
```

Comment:
It also would be very useful for debugging purpose to check what is in `rendered`. And not only in case of template error. Template syntax might be fine, but produce different output. Write it to stdout? I am ok, if you do it in separate PR.

Reply:
Agreed to put this into a `tdbg` command for now, and later move into the server binary once we're happy (due to backwards compat requirements).